solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -69,119 +145,317 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
         return
-            _dispatch(
-                _destinationDomain,
-                InterchainCallMessage.format(
-                    calls,
-                    msg.sender.addressToBytes32()
-                )
+            getLocalInterchainAccount(
+                _origin,
+                _routerAsBytes32,
+                _ownerAsBytes32,
+                _ism
             );
     }
 
     /**
-     * @notice Returns the address of the interchain account deployed on the current chain for a given `origin`/`sender` pair.
-     * @param _origin The origin domain of the interchain account.
-     * @param _sender The parent account address on the origin domain.
-     * @return The address of the interchain account.
+     * @notice Returns the domains which have default remote routers enrolled
+     * @dev This does not include domains which may have user-specific overrides
+     * @return The domains which have default remote routers enrolled
      */
-    function getInterchainAccount(uint32 _origin, bytes32 _sender)
+    function domains() external view returns (uint32[] memory) {
+        bytes32[] storage rawKeys = OverridableEnumerableMap.keysDefault(
+            routersMap
+        );
+        uint32[] memory keys = new uint32[](rawKeys.length);
+        for (uint256 i = 0; i < rawKeys.length; i++) {
+            keys[i] = uint32(uint256(rawKeys[i]));
+        }
+        return keys;
+    }
+
+    /**
+     * @notice Returns the default router address for a remote domain
+     * @param _domain The remote domain
+     */
+    function routers(uint32 _domain) external view returns (bytes32) {
+        if (OverridableEnumerableMap.containsDefault(routersMap, _domain)) {
+            return OverridableEnumerableMap.getDefault(routersMap, _domain);
+        } else {
+            return bytes32(0); // for backwards compatibility with storage mapping
+        }
+    }
+
+    // ============ Public Functions ============
+
+    /**
+     * @notice Registers the address of a remote InterchainAccountRouter
+     * contract to use as a default when making interchain calls
+     * @param _destination The remote domain
+     * @param _router The address of the remote InterchainAccountRouter
+     */
+    function enrollRemoteRouter(uint32 _destination, bytes32 _router) public {
+        enrollRemoteRouterAndIsm(_destination, _router, bytes32(0));
+    }
+
+    /**
+     * @notice Registers the address of remote InterchainAccountRouter
+     * and ISM contracts to use as a default when making interchain calls
+     * @param _destination The remote domain
+     * @param _router The address of the remote InterchainAccountRouter
+     * @param _ism The address of the remote ISM
+     */
+    function enrollRemoteRouterAndIsm(
+        uint32 _destination,
+        bytes32 _router,
+        bytes32 _ism
+    ) public onlyOwner {
+        require(_router != bytes32(0), "invalid router address");
```

Comment:
Also seems to be missing a zero check on the ism

Reply:
Ditto, the intention was to allow ISM to be zero to defer to the Mailbox default ISM.

Maybe I should add the routing ISM to this PR?

Reply:
see https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1832/files#r1134027231

Remind me what the ICA routing ISM looks like?

Reply:
It defers to the ISM specified in the InterchainAccountMessage